### ISSUE TITLE
Implement canonical data for all non-deprecated exercises

### ISSUE THREAD

**Issue by kytrinyx** [exercism/x-common#552]
A number of the exercises are still missing **canonical data**, a standardized set of test inputs/outputs that makes it easier to [implement exercises across multiple language tracks][porting] and [keep exercises consistent][improving-consistency] as the test suites evolve.

These issues link back to this one. Once they're all fixed and closed, this issue can also be closed.

[porting]: https://github.com/exercism/exercism.io/blob/master/docs/porting-an-exercise.md
[improving-consistency]: https://github.com/exercism/exercism.io/blob/master/docs/improving-consistency-across-tracks.md

**Comment by petertseng** [comment]
I have some specific thoughts about this problem.

Most problems you will see are stateless - write a function, give it some inputs, expect a certain output. All exercises listed in the example section in fact follow that pattern (custom-set is a bit more complex, but fundamentally it still fits into that pattern, just with multiple functoins)

You may find that bank-account doesn't fall into that pattern if it is very stateful - create a bank account, do some things to that bank account, check some state of that bank account.

If it looks like the bank-account tests can't fit well with the examples listed, try looking at what happened in https://github.com/exercism/x-common/pull/488 and https://github.com/exercism/x-common/pull/358 , which were also two very stateful exercises.

(Of course, if you find that bank-account can fit well into the stateless model, then you can ignore everything I've said, no worries!)

**Comment by petertseng** [comment]
Sigh, I aimed at the wrong place. I'll try again.

**Comment by stkent** [comment]
Should canonical data be required for any new exercises added (if it isn't already)?

**Comment by kytrinyx** [comment]
That's a good question. I think that it should be encouraged, but another option is to open a "canonical data" issue for the new exercise and let someone else tackle it.

**Comment by kotp** [comment]
Given a "Proof of Concept" as the given example program and tests, the canonical is probably not a leap for a second party to tackle, and, presumably, the creator is interested and so still available for input.  I think that works.

**Comment by petertseng** [comment]
Don't mind me, just putting this somewhere where I can find it easily again:

``` bash
curl http://x.exercism.io/v3/problems/data-todos | jq '.[] | map(select(.["slug"] == "accumulate"))'
```

**Comment by petertseng** [comment]
Hi. I'm going to mark all the subtasks of this task as good first patch.

**Comment by m-a-ge** [comment]
Both links mentioned in the description are dead now:
* ~~[implement exercises across multiple language tracks](https://github.com/exercism/exercism.io/blob/master/docs/porting-an-exercise.md)~~
* ~~[keep exercises consistent](https://github.com/exercism/exercism.io/blob/master/docs/improving-consistency-across-tracks.md)~~

I assume the valid one should be **[Getting Involved in an Exercism Language Track](https://github.com/exercism/docs/tree/master/contributing-to-language-tracks)** to avoid any confusion for newcomers. 
I also think there is no point of updating any sub issues as long as the parent has correct info

@kytrinyx, could you please update the description, here is the link:
`[Getting Involved in an Exercism Language Track](https://github.com/exercism/docs/tree/master/contributing-to-language-tracks)`

**Comment by kytrinyx** [comment]
Good call, @m-a-ge, I've rewritten a bit to add the suggested link, and fix the two dead ones.

**Comment by ErikSchierboom** [comment]
I've just ran a command (`find . -type d -maxdepth 1 -mindepth 1 -exec ls {}/canonical-data.json \; | grep 'No such'
`) to check which canonical data are not yet implemented:

```
ls: ./error-handling/canonical-data.json: No such file or directory
ls: ./ledger/canonical-data.json: No such file or directory
ls: ./strain/canonical-data.json: No such file or directory
ls: ./dot-dsl/canonical-data.json: No such file or directory
ls: ./lens-person/canonical-data.json: No such file or directory
ls: ./tree-building/canonical-data.json: No such file or directory
ls: ./hangman/canonical-data.json: No such file or directory
ls: ./paasio/canonical-data.json: No such file or directory
ls: ./point-mutations/canonical-data.json: No such file or directory
ls: ./nucleotide-codons/canonical-data.json: No such file or directory
ls: ./bank-account/canonical-data.json: No such file or directory
ls: ./robot-name/canonical-data.json: No such file or directory
ls: ./counter/canonical-data.json: No such file or directory
ls: ./hexadecimal/canonical-data.json: No such file or directory
ls: ./octal/canonical-data.json: No such file or directory
ls: ./simple-linked-list/canonical-data.json: No such file or directory
ls: ./parallel-letter-frequency/canonical-data.json: No such file or directory
```
